apps/web/src/stores/editor-store.ts

```diff
@@ -676,8 +680,9 @@ class EditorStore extends BaseStore<EditorStore> {
     const oldTabForNote = options.force ? null : getTabsForNote(noteId).at(0);
     const activeTab = getActiveTab();
     const tabId =
-      // if a tab is pinned then always open a new tab.
-      options.openInNewTab || activeTab?.pinned
+      options.openInNewTab ||
+      // if a tab is pinned then always open a new tab, unless there's an existing tab for the note
+      (options.considerPinnedTab && activeTab?.pinned && !oldTabForNote)
```

Comment:
Another way of doing it could be by always defining the `oldTabForNote` variable. This does fix the pinned tab not opening bug, but introduces another one: if there is a note open in two tabs, trying to open the second tab will always open the first tab (this occurs after reloading the page)

```ts
    const oldTabForNote = getTabsForNote(noteId).at(0);
    const activeTab = getActiveTab();
    const tabId =
      options.openInNewTab ||
      (activeTab?.pinned && !oldTabForNote)
```

^ is this approach worth exploring, @thecodrr?